osu.Game/Screens/Select/BeatmapInfoWedgeV2.cs

```diff
@@ -0,0 +1,346 @@
+                Radius = corner_radius,
+                Roundness = corner_radius
+            };
+            CornerRadius = corner_radius;
+
+            // We want to buffer the wedge to avoid weird transparency overlaps between the colour bar and the background.
+            Child = bufferedContent = new BufferedContainer
+            {
+                RelativeSizeAxes = Axes.Both,
+                Children = new Drawable[]
+                {
+                    // These elements can't be grouped with the rest of the content, due to being present either outside or under the backgrounds area
+                    difficultyColourBar = new Container
+                    {
+                        Colour = Colour4.Transparent,
+                        Depth = float.MaxValue,
+                        Anchor = Anchor.TopRight,
+                        Origin = Anchor.TopRight,
+                        RelativeSizeAxes = Axes.Y,
+
+                        // By limiting the width we avoid this box showing up as an outline around the drawables that are on top of it.
+                        Width = colour_bar_width + corner_radius,
+                        Child = new Box { RelativeSizeAxes = Axes.Both }
+                    },
+                    starCounter = new StarCounter
+                    {
+                        Colour = Colour4.Transparent,
+                        Anchor = Anchor.CentreRight,
+                        Origin = Anchor.Centre,
+                        Scale = new Vector2(0.35f),
+                        Shear = -wedged_container_shear,
+                        X = -colour_bar_width / 2,
+                        Direction = FillDirection.Vertical
+                    }
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            ruleset.BindValueChanged(_ => updateDisplay());
+
+            float starAngle = (float)(Math.Atan(shear_width / wedge_height) * (180 / Math.PI));
+
+            // Applying the rotation directly to the StarCounter distorts the stars, hence it is applied to the child container
+            starCounter.Children.First().Rotation = starAngle;
+
+            // Makes sure the stars center themselves properly in the colour bar
+            starCounter.Children.First().Anchor = Anchor.Centre;
+            starCounter.Children.First().Origin = Anchor.Centre;
+        }
+
+        private const double animation_duration = 800;
+
+        protected override void PopIn()
+        {
+            this.MoveToX(0, animation_duration, Easing.OutQuint);
+            this.RotateTo(0, animation_duration, Easing.OutQuint);
+            this.FadeIn(transition_duration);
+        }
+
+        protected override void PopOut()
+        {
+            this.MoveToX(-100, animation_duration, Easing.In);
+            this.RotateTo(10, animation_duration, Easing.In);
+            this.FadeOut(transition_duration * 2, Easing.In);
+        }
+
```

Comment:
This will need reconsideration. The animation was made by me for the previous version, and does not look correct with this one.

Reply:
i considered this more of a change to be made during integration, since it'd be easier to see what would fit. but sure ill address